### PR TITLE
Self-hosted user UI

### PR DESCRIPTION
**Pull request by nbradbury**
In this PR, I've combined #21264 (self-hosted user list) and #21270 (self-hosted user detail). This did make for a larger PR, but it should be easier to review than having separate PRs, plus a lot of the code is Jetpack Compose which can be quite verbose.

To test:

- Enable `self_hosted_users` in debug settings
- Switch to a self-hosted site
- Note that a new "Users" item appears
- Tap on that and verify the list screen with dummy data looks fine
- Tap a user to view the detail screen and verify that looks fine
- If the user has an avatar, tap it and verify the large avatar view looks fine
- Switch to a non-self-hosted site and verify the "Users" item doesn't appear
- View the various Compose previews of the `SelfHostedUsersScreen` and verify they look fine (note that remote avatar images won't load in the preview)

Caveats:

- This is a UI-only PR - no network requests are made, instead only dummy data is used
- The UI is somewhat different than in [this iOS PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/23572) but it's closer to the appearance of the existing People feature on Android

https://github.com/user-attachments/assets/14686212-5976-44cf-abcb-a84238973fbf

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          5 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>SelfHostedUsersViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>SelfHostedUserState</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/items/listitem/SiteListItemBuilder.kt [hunk 1]
```diff
@@ -138,6 +141,20 @@ class SiteListItemBuilder @Inject constructor(
                 onClick = ListItemInteraction.create(PEOPLE, onClick),
                 listItemAction = PEOPLE
             )
+        } else {
+            null
+        }
+    }
+
+    fun buildSelfHostedUserListItemIfAvailable(site: SiteModel, onClick: (ListItemAction) -> Unit): ListItem? {
```
Comment (dcalhoun) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/items/listitem/SiteListItemBuilder.kt:149]: From my perspective, the screen reader experience for this new list could be improved. E.g., swiping up/down moves between individual user items, but merely announces "list" repeatedly, which provides little context. 

We might improve the experience by having the top level list declare the total number of items ("list, 25 of 35 users"), individual items could announce some high-level information ("dcalhoun, administrator"), etc. 

I perceive this file is shared amongst several lists. I'm uncertain if that abstraction might make these types of improvements  difficult, but I share this feedback nonetheless.

File: WordPress/src/main/java/org/wordpress/android/ui/selfhostedusers/SelfHostedUserComposables.kt [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+/**
+ * A composable that displays a message when there is no network connection
+ */
+@Composable
+fun OfflineView() {
+    MessageView(
+        R.drawable.ic_wifi_off_24px,
+        R.string.no_network_title,
+        R.string.no_network_message,
```
Comment (dcalhoun) [inline WordPress/src/main/java/org/wordpress/android/ui/selfhostedusers/SelfHostedUserComposables.kt:121]: I presume this was an existing string and unrelated to the proposed changes, but I feel displaying "No network available" (title) and "There is no network available" (message) next to each other is redundant and unhelpful. 

If we wanted to improve the messaging, we might consider updating the message to say "Try again after reconnecting to the internet." However, if we do update the string, we should verify it makes sense for all current usages.

**Review by dcalhoun — APPROVED** [review]
Nice! Exciting to see this feature begin to take shape. The testing instructions passed for me using a Samsung Galaxy S20 running Android 13. 

I left a few suggestions to consider, but I do not consider them blockers. Let me know what you think.

File: WordPress/src/main/java/org/wordpress/android/ui/selfhostedusers/SelfHostedUserComposables.kt (hunk 2 above)
Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/selfhostedusers/SelfHostedUserComposables.kt:121]: > I presume this was an existing string and unrelated to the proposed changes, but I feel displaying "No network available" (title) and "There is no network available" (message) next to each other is redundant and unhelpful.

I agree here, but you're correct I was using existing wording used elsewhere in the app. I filed #21283 to address this.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/items/listitem/SiteListItemBuilder.kt (hunk 1 above)
Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/items/listitem/SiteListItemBuilder.kt:149]: @dcalhoun Thanks for calling this out! I ended up filing issue #21284 to tackle this. I've not worked much with accessibility in Jetpack Compose so chose to tackle that separately.

**Comment by nealgoogs** [comment]
Please take me off the email list. I don't know why I'm getting this GitHub notifications. I'm not a part of this group.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Review by dcalhoun — APPROVED** [review]

**Pull request merged**